lib/Conversion/TorchToMhlo/Reduction.cpp

```diff
@@ -556,6 +569,106 @@ LogicalResult ConvertAtenReductionOp<AtenSumDimIntListOp>::matchAndRewrite(
 }
 } // namespace
 
+// AtenFrobeniusNormDimOp
+// aten.frobenius_norm.dim => mhlo.reduce(calculate square sum along given dims)
+//                            + mhlo.sqrt
+namespace {
+template <>
+LogicalResult ConvertAtenReductionOp<AtenFrobeniusNormDimOp>::matchAndRewrite(
+    AtenFrobeniusNormDimOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  Value input = adaptor.self();
+  auto inputType = input.getType().dyn_cast<RankedTensorType>();
+  if (!inputType) {
+    return op.emitError(
+        "only ranked tensor input supported in AtenFrobeniusNormDimOp");
+  }
+  auto inputRank = inputType.getRank();
+  auto inputElemType = inputType.getElementType();
+  if (!inputElemType.isa<mlir::FloatType>()) {
+    return op.emitError(
+        "only float dtype allowed in input tensor of AtenFrobeniusNormDimOp");
+  }
+
+  SmallVector<int64_t> dims;
+  if (!matchPattern(op.dim(), m_TorchConstantIntList(dims))) {
+    return rewriter.notifyMatchFailure(
+        op, "non-const integer `dim` is not supported");
+  }
+  for (auto &dim : dims) {
```

Comment:
Nit: How about sorting the dimensions in ascending order so that the conversion can be stable with unordered dims inputs?

Reply:
Done.